app/controllers/users.js

```diff
@@ -185,4 +186,35 @@ exports.user = function(req, res, next, id) {
       req.profile = user;
       next();
     });
-};
\ No newline at end of file
+};
+
+
+exports.search = (req, res) => {
+  // search users with query ==>
+  const { user } = req.query;
+
+  const queryObject = {};
+  // check if there is a query at all . 
+  if (!user || user && typeof parseInt(user) === 'int') {
+    return res.status(422).json({
+      message: 'No user search query provided.'
+    });
+  }
+
+  isEmail(user) ? queryObject.email = user : queryObject.name = user;
+
+  User.find(queryObject, (error, users) => {
+    if (error) {
+      console.log(error);
+    }
+
+    if (users.length > 0) {
+      const userResponse = users.map(({ name, email, avatar, username }) => ({
+        name, email, avatar, username
+      }));
+      return res.json({ users: userResponse });      
+    } else {
```

Comment:
No use of `else` here. You could simply have `return res.status(404).json({ message: ...});`. Gives you the same result.